src/emulator/portUtils.ts

```diff
@@ -1,9 +1,12 @@
 import * as pf from "portfinder";
 import * as tcpport from "tcp-port-used";
+import * as dns from "dns";
 
 import { FirebaseError } from "../error";
 import { logger } from "../logger";
 
+dns.setDefaultResultOrder("ipv4first");
```

Comment:
Okay I don't love this idea either, but can we feature detect this?
```suggestion
if (dns.setDefaultResultOrder) {
  dns.setDefaultResultOrder("ipv4first");
}
```

Reply:
We _can_, but I've also updated the engines here to `v14.18`, where this function was added. Do you still think it's worth having the check?

For what it's worth, I've discovered that a dependency does `require('node:process')`, which support for was only added in node 14.18 anyway, so... if it breaks, it'd break in glorious fashion.